src/config/settings.py

```diff
@@ -0,0 +1,56 @@
+# src/config/settings.py
+from pathlib import Path
+from pydantic import  Field
+from pydantic_settings import BaseSettings
+BASE_DIR = Path(__file__).resolve().parent.parent.parent
+
+class AppConfig(BaseSettings):
+    """
+    Application configuration class that loads and validates settings from environment variables.
+
+    This class uses Pydantic's BaseSettings to load configuration from environment variables
+    and validate their types. It defines paths to various resources used by the application,
+    such as database files, input/output folders, and SQL query files.
+    """
+
+    data_date: str = Field(..., validation_alias="DATA_DATE")
+    db_path: Path = Field(..., validation_alias="DB_PATH")
+    funds_folder: Path = Field(..., validation_alias="FUNDS_FOLDER")
+    report_output_folder: Path = Field(..., validation_alias="REPORT_OUTPUT_FOLDER")
+    report_recon_path: Path = Field(..., validation_alias="REPORT_RECON_PATH")
+    report_perf_path: Path = Field(..., validation_alias="REPORT_PERF_PATH")
+    tbl_raw_funds_details: str = Field(..., validation_alias="TBL_RAW_FUNDS_DETAILS")
+
+    sql_query_master_reference: Path = Field(..., validation_alias="SQL_QUERY_MASTER_REFERENCE")
+    sql_query_base_tables: Path = Field(..., validation_alias="SQL_QUERY_BASE_TABLES")
+    sql_query_get_active_funds_cfg: Path = Field(..., validation_alias="SQL_QUERY_GET_ACTIVE_FUNDS_CFG")
+    sql_query_get_raw_reference_data: Path = Field(..., validation_alias="SQL_QUERY_GET_RAW_REFERENCE_DATA")
+    sql_query_get_pub_reference_data: Path = Field(..., validation_alias="SQL_QUERY_GET_PUB_REFERENCE_DATA")
+    sql_query_get_pub_funds_equities_data: Path = Field(..., validation_alias="SQL_QUERY_GET_PUB_FUNDS_EQUITIES_DATA")
+
```

Comment:
These are all good candidates for pydantic attributes though a more typical Pythonic style would be to have the attributes associated with the corresponding pipeline stage (i.e. a class) instead of only within a config object that acts as a bag of values.